### PR TITLE
[iOS] Disable additional sourcegenerator tests

### PR DESCRIPTION
**Pull request by steveisok**
These aren't necessary to run on mobile targets, so skip them.

### CODE REVIEW

File: src/libraries/tests.proj [hunk 1]
```diff
@@ -99,6 +99,9 @@
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Unit.Tests\System.Text.Json.SourceGeneration.Roslyn4.0.Unit.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn4.0.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Unit.Tests\System.Text.Json.SourceGeneration.Roslyn4.4.Unit.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn4.4.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.RegularExpressions\tests\FunctionalTests\System.Text.RegularExpressions.Tests.csproj" />
```
Comment (mdh1418) [inline src/libraries/tests.proj:104]: Why is this included in skipping SourceGeneration tests?
Is this also significantly expensive and contributing to timeouts?

File: src/libraries/tests.proj [hunk 2]
```diff
@@ -99,6 +99,9 @@
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Unit.Tests\System.Text.Json.SourceGeneration.Roslyn4.0.Unit.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn4.0.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Unit.Tests\System.Text.Json.SourceGeneration.Roslyn4.4.Unit.Tests.csproj" />
+    <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn4.4.Tests.csproj" />
```
Comment (mdh1418) [inline src/libraries/tests.proj:103]: These are the same projects as the `4.0.Tests`, they were just renamed in https://github.com/dotnet/runtime/pull/79828. We should either use regex to match the projects via something like `System.Text.Json\tests\System.Text.Json.SourceGeneration.*\**\*.csproj` or disable the tests in the projects themselves as suggested by Alex, similar to https://github.com/dotnet/runtime/blob/6251d496e307ed39f1e600cd036106cc1bd7e340/src/libraries/System.Drawing.Common/tests/System.Drawing.Common.Tests.csproj#L6

File: src/libraries/tests.proj (hunk 1 above)
Comment (steveisok) [reply src/libraries/tests.proj:104]: Yes, they all take too long and it's questionable if they are necessary to run on mobile targets.

**Review by mdh1418 — APPROVED** [review]
If the tests are properly skipped on CI, looks good to me

**Comment by steveisok** [comment]
All ios failures are related to https://github.com/dotnet/arcade/issues/11683
